src/Uno.UI/UI/Xaml/Controls/Image/Image.iOS.cs

```diff
@@ -259,28 +261,53 @@ private void UpdateContentMode(Stretch stretch)
 							"Stretch mode {0} is not supported".InvariantCultureFormat(stretch));
 				}
 			}
+			else
+			{
+				SetNeedsLayout();
+			}
 		}
 
+		public override void LayoutSubviews()
+		{
+			try
+			{
+				_layouter.Arrange(Bounds);
+
+				UpdateLayerRect(Bounds.Size);
+			}
+			catch (Exception e)
+			{
+				this.Log().Error($"Layout failed in {GetType()}", e);
+			}
+		}
 
 		partial void OnStretchChanged(Stretch newValue, Stretch oldValue)
 		{
 			UpdateContentMode(newValue);
 		}
 
+		private CGSize _previousSize;
+
 		public override CGSize SizeThatFits(CGSize size)
 		{
 			size = IFrameworkElementHelper.SizeThatFits(this, size);
 
 			size = _layouter.Measure(size.ToFoundationSize());
 
-			UpdateLayerRect(size);
+			if (size != _previousSize)
+			{
+				SetNeedsLayout();
```

Comment:
It's kinda strange to call `SetNeedsLayout()` from `SizeThatFits()`

Reply:
You're right, it's an artefacts from tests.  Good catch.